pkg/kgo/compression.go

```diff
@@ -24,31 +24,41 @@ func (s *sliceWriter) Write(p []byte) (int, error) {
 
 var sliceWriters = sync.Pool{New: func() interface{} { r := make([]byte, 8<<10); return &sliceWriter{inner: r} }}
 
+type codecType int8
+
+const (
+	codecNone = iota
```

Comment:
These consts should have type `codecType`.

Reply:
Whoops, good catch. Opened https://github.com/twmb/franz-go/pull/270 .

Reply:
@PleasingFungus and I work together.

Reply:
Awesome! Happy to see a prominent member of the Go community in my humble repo 😅